category.go

```diff
@@ -1,49 +1,56 @@
 package cli
 
-// CommandCategories is a slice of *CommandCategory.
+// CommandCategories wraps a slice of *CommandCategory.
 type CommandCategories struct {
-	Categories []*CommandCategory
+	slice []*CommandCategory
 }
 
 func NewCommandCategories() *CommandCategories {
-	return &CommandCategories{Categories: []*CommandCategory{}}
-}
-
-// CommandCategory is a category containing commands.
-type CommandCategory struct {
-	Name     string
-	Commands []*Command
+	return &CommandCategories{slice: []*CommandCategory{}}
 }
 
 func (c *CommandCategories) Less(i, j int) bool {
-	return c.Categories[i].Name < c.Categories[j].Name
+	return c.slice[i].Name < c.slice[j].Name
 }
 
 func (c *CommandCategories) Len() int {
-	return len(c.Categories)
+	return len(c.slice)
 }
 
 func (c *CommandCategories) Swap(i, j int) {
-	c.Categories[i], c.Categories[j] = c.Categories[j], c.Categories[i]
+	c.slice[i], c.slice[j] = c.slice[j], c.slice[i]
 }
 
-// AddCommand adds a command to a category.
-func (c *CommandCategories) AddCommand(category string, command *Command) *CommandCategories {
-	for _, commandCategory := range c.Categories {
+// AddCommand adds a command to a category, creating a new category if necessary.
+func (c *CommandCategories) AddCommand(category string, command *Command) {
```

Comment:
Maybe `command` should be variadic here? Would ease appending a list of commands.